src/core.c/Main.pm6

```diff
@@ -107,7 +111,12 @@ my sub RUN-MAIN(&main, $mainline, :$in-as-argsfiles) {
                       !! thevalue(nqp::join("=",$split));
                 }
 
-                # implicit value
+                # explicit value (option) using whitespace
+                elsif $arg ∈ @script-options && !@args[0].starts-with('-'|':') {
```

Comment:
If this is running at startup and we really want to be concerned about performance, Junctions are very slow. I'd recommend `&& !(@args[0].starts-with('-') || @args[0].starts-with(':'))`

Reply:
Thanks!  I've actually been refactoring this PR quite a bit to address the more substantive issues raised above (and avoid any incompatible changes).  So this line won't be in the PR. and (I think) the new version should generally be faster.  But I appreciate the tip and will keep an eye out for similar performance issues. :+1: